pkg/roachprod/logger/log.go

```diff
@@ -195,6 +196,19 @@ func (l *Logger) Closed() bool {
 	return l.mu.closed
 }
 
+// RootLogger returns the logger instance without a parent in this
+// logger chain. If the logger instance was not created with
+// `ChildLogger`, this function will return the a reference to
+// receiver itself.
+func (l *Logger) RootLogger() *Logger {
```

Comment:
What ensures that `l.parent` is actually the _root_ of the chain?

Reply:
Not sure I understand the question: `l.parent` may not be the root of the chain, which is why we iterate until `l.parent == nil` in this function. Could you clarify?

Reply:
> which is why we iterate

Right. Upon last reading, somehow I missed the loop and read it as one step dereference of the chain.